core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -49,6 +53,8 @@ public class Deletes {
   private static final Accessor<StructLike> POSITION_ACCESSOR =
       POSITION_DELETE_SCHEMA.accessorForField(MetadataColumns.DELETE_FILE_POS.fieldId());
 
+  private static ExecutorService deletePosThreadPool = ThreadPools.newDeleteWorkerPool();
```

Comment:
I think that this default pool should be managed by ThreadPools like the worker pool, rather than using a ThreadPools factory method to create them. See my comments in that class.